File: OrganizationService/src/main/java/com/slavacom/organizationservice/entity/ProjectEmployees.java
package com.slavacom.organizationservice.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "project_employees")
public class ProjectEmployees {
	@Id
	@GeneratedValue(strategy = GenerationType.UUID)
	@Column(name = "id", nullable = false)
	private UUID id;

	@Column(name = "project_id", nullable = false)
	private UUID projectId;

	@Column(name = "user_id", nullable = false)
	private UUID userId; // legacy ID пользователя для совместимости с текущей схемой БД

	@Column(name = "profile_id", nullable = false)
	private UUID profileId; // ID профиля из User Service

	@Column(name = "role", nullable = false)
	private String role; // роль в проекте (LEAD, DEVELOPER, TESTER, etc.)

	@Column(name = "permissions", columnDefinition = "TEXT")
	private String permissions; // JSON строка с правами доступа в проекте

	@Column(name = "is_active", nullable = false)
	private boolean isActive = true;

	@CreationTimestamp
	@Column(name = "created_at", nullable = false, updatable = false)
	private Instant createdAt;

	@UpdateTimestamp
	@Column(name = "updated_at")
	private Instant updatedAt;
}

File: OrganizationService/src/main/java/com/slavacom/organizationservice/entity/ProjectEmployees.kt
package com.slavacom.organizationservice.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "project_employees")
open class ProjectEmployees {
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.UUID)
    @field:Column(name = "id", nullable = false)
    open var id: UUID? = null

    @field:Column(name = "project_id", nullable = false)
    open var projectId: UUID? = null

    @field:Column(name = "user_id", nullable = false)
    open var userId: UUID? = null

    @field:Column(name = "profile_id", nullable = false)
    open var profileId: UUID? = null

    @field:Column(name = "role", nullable = false)
    open var role: String? = null

    @field:Column(name = "permissions", columnDefinition = "TEXT")
    open var permissions: String? = null

    @field:Column(name = "is_active", nullable = false)
    open var isActive: Boolean = true

    @field:CreationTimestamp
    @field:Column(name = "created_at", nullable = false, updatable = false)
    open var createdAt: Instant? = null

    @field:UpdateTimestamp
    @field:Column(name = "updated_at")
    open var updatedAt: Instant? = null
}

